### PR TITLE
Add archive param for proxmox_kvm module

### PR DESCRIPTION
**Pull request by l00ptr**
##### SUMMARY
We  add a param to allow proxmox administrator to restore VM from an archives. This new option will be useful to automate automatic creation and restore of VM.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
proxmox_kvm

### CODE REVIEW

File: plugins/modules/proxmox_kvm.py [hunk 1]
```diff
@@ -1306,6 +1320,7 @@ def main():
                               sshkeys=module.params['sshkeys'],
                               startdate=module.params['startdate'],
                               startup=module.params['startup'],
+                              storage=module.params['storage'],
```
Comment (felixfontein) [inline plugins/modules/proxmox_kvm.py:1323]: This change seems unrelated. Is this a bugfix? If yes, it should be a separate PR.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

File: plugins/modules/proxmox_kvm.py [hunk 2]
```diff
@@ -1053,6 +1065,7 @@ def stop_vm(self, vm, force):
 def main():
     module_args = proxmox_auth_argument_spec()
     kvm_args = dict(
+        archive=dict(type='str'),
```
[the remark this answers is not in the document]
Comment (l00ptr) [reply plugins/modules/proxmox_kvm.py:1068]: archive can be something like `backup-storage:backup/vm/140/2023-03-08T06:41:23Z` is it a valid path ? I guess the type is used for some typing validation (and check if the value is valid path or not) in such case we should keep the str type ?

File: plugins/modules/proxmox_kvm.py (hunk 1 above)
Comment (l00ptr) [reply plugins/modules/proxmox_kvm.py:1323]: indeed it's not strictly mandatory for this change, but we probably need to allow user to define the storage target when creating  or  restoring a VM. I will remove this on the current PR & create a separate one!

File: plugins/modules/proxmox_kvm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/proxmox_kvm.py:1068]: No, that's not a valid path.

[the remark this answers is not in the document]
Comment (l00ptr) [reply plugins/modules/proxmox_kvm.py:1068]: i don't know if it's possible to write our own validator or accepting path or valid archive path. We can probably check the value of archive argument and throw and error if it doesn't respect what proxmox except, do you think we should implement this or keep a simple str without validation ?

[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/modules/proxmox_kvm.py:1068]: I would simply accept a string. Writing your own validator is somewhat dangerous in case proxmox suddenly extends the syntax, then the code needs to be adjusted.

File: changelogs/fragments/6158-create-proxmox-vm-from-archive.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
```
Comment (felixfontein) [inline changelogs/fragments/6158-create-proxmox-vm-from-archive.yml:1]:
```suggestion
minor_changes:
```

**Review by felixfontein — COMMENTED** [review]
If nobody objects, I'll mege this in a week (with my suggestion).

**Comment by felixfontein** [comment]
@l00ptr thanks a lot for your contribution!
